File: cpp/problem521.cpp
#include <string>
#include <iostream>
using namespace std;

// Class: Ordinary Solution.
// Description:
//  - this is too easy.
//  - constant space and time costs.
// Creat: Yangyao, date 20181004
// References: none.
class Solution {
public:
    int findLUSlength(string a, string b) {
        if( a.size() != b.size() ){
            return max( a.size(), b.size() );
        }else if( a == b ){
            return -1;
        }else{
            return a.size();
        }
    }
};


// Func: Driver function to test solutions
int main(int argc, char const *argv[]){

    Solution sol;

    // Test tree cases that cover all situations.
    cout << sol.findLUSlength("abc","cba") << ", expect 3" << endl;
    cout << sol.findLUSlength("abc","abc") << ", expect -1" << endl;
    cout << sol.findLUSlength("abc","abcd") << ", expect 4" << endl;

    return 0;
}

File: python/problem521.py
class Solution(object):
    def findLUSlength(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: int
        """
        if (not a and b)or (not b and a):
            return max(len(a),len(b))
        if a==b:
            return -1
        a1=[]
        b1=[]
        count=[]
        for i in range(len(a)):
            for k in range(i+1,len(a)+1):
                a1.append(a[i:k])
        for i in range(len(b)):
            for k in range(i+1,len(b)+1):
                b1.append(b[i:k])
        for i in range(len(a1)):
            if a1[i] not in b:
                count.append(len(a1[i]))
        for i in range(len(b1)):
            if b1[i] not in a:
                count.append(len(b1[i]))
        return max(count)
        
        
        
        
        
# from discuss
'''
def findLUSlength(self, a, b):
    """
    :type a: str
    :type b: str
    :rtype: int
    """
    if a == b:
        return -1
    else:
        return max(len(a), len(b))
'''
